accepted/2023/simplify-output-paths.md

```diff
@@ -0,0 +1,130 @@
+- `bin\<Configuration>\<TargetFramework>\publish` - Publish output with no `RuntimeIdentifier`
+- `bin\<Configuration>\<TargetFramework>\<RuntimeIdentifier>\publish` - Publish output with `RuntimeIdentifier`
+
+## Proposed behavior
+
+We will introduce a new output path format, called the "artifacts" output format after the default folder name it will use.  By convention, the artifacts folder will be located at the solution or repo level rather within each project folder.
+
+The recommended way to opt in to the new output path format will be to set the `UseArtifactsOutput` property to `true` in a `Directory.Build.props` file.
+
+The artifacts folder will by default be named `.artifacts`.  The convention we will use to determine which folder the artifacts folder should go in will be:
+
+- If a `Directory.Build.props` file was used in the build, put the artifacts folder in the same folder where `Directory.Build.props` was found, which will typically be at the root of a repo.  Since MSBuild already probes for `Directory.Build.props`, we should be able to use this folder for the artifacts with minimal perf impact
+- Look for a folder with a `.sln` file in it, walking up the directory tree from the current project folder to the root.  When a `.sln` file is found, put the artifacts folder in that same directory.  We may need an additional intrinsic MSBuild function to do this, as the `GetDirectoryNameOfFileAbove` function doesn't support wildcards for the file name
+- If neither a `Directory.Build.props` file nor a `.sln` file was found, then put the artifacts folder in the project folder.
+
+To override the default artifacts folder path, the `ArtifactsPath` property can be used.  If `ArtifactsPath` is set, then `UseArtifactsOutput` will default to `true` and it won't be necessary to set both properties.  `ArtifactsPath` should normally be set in a `Directory.Build.props` file.  For example:
+
+```xml
+<Project>
+    <PropertyGroup>
+        <ArtifactsPath>$(MSBuildThisFileDirectory)\output</ArtifactsPath>
+    </PropertyGroup>
+</Project>
+```
+
+Under the artifacts folder will be nested folders for the type of the output, the project name, and the pivots, ie `<ArtifactsPath>\<Type of Output>\<Project Name>\<Pivots>`.
+
+- Type of Output - This will be a value such as `bin`, `publish`, `intermediates`, or `package`.  Projects will be able to override this value, for example to separate shipping and non-shipping artifacts into different folders
+  - `bin` will be the folder for the normal build output.  Projects can override this with `ArtifactsOutputName`
+  - `publish` will be the folder for the publish output.  Projects can override this with `ArtifactsPublishOutputName`
+  - `package` will be the folder for the package output, where .nupkg folders are placed. (Or should it be `packages`?  Feedback welcome.)  Projects can override this with `ArtifactsPackageOutputName`
```

Comment:
`ArtifactsOutputName` looks inconsistent with the rest and lacking meaning, propose `ArtifactsBinOutputName` or something similar that indicates this is a folder where binaries go.